sample/sample.go

```diff
@@ -21,10 +22,16 @@ type SamplerFactory struct {
 	Metrics metrics.Metrics `inject:"metrics"`
 }
 
-// GetSamplerImplementationForDataset returns the sampler implementation for the dataset,
-// or nil if it is not defined
-func (s *SamplerFactory) GetSamplerImplementationForDataset(dataset string) Sampler {
-	c, err := s.Config.GetSamplerConfigForDataset(dataset)
+// GetSamplerImplementationForKey returns the sampler implementation for the given
+// samplerKey (dataset or environment), or nil if it is not defined
+func (s *SamplerFactory) GetSamplerImplementationForKey(samplerKey string, isLegacyKey bool) Sampler {
+	if isLegacyKey {
+		if prefix := s.Config.GetDatasetPrefix(); prefix != "" {
+			samplerKey = fmt.Sprintf("%s.%s", prefix, samplerKey)
```

Comment:
You could mandate that the intervening period be part of the specified prefix. If I set the `DatasetPrefix` field to "x," the prefix really becomes "x."—even though I didn't mention the period when specifying my preferred prefix. I might prefer a prefix like "dataset-" (with a separating hyphen instead of a period).

Had you considered including this separator as part of the prefix? Were you attempting to avoid false aliasing if there was no obvious separator character included in the prefix?

Reply:
Yes, I agree additional docs on the prefix format where the `.` separator is added for you would be useful 👍🏻 

The separator is not necessary but I think helps create a mental divide. The intent of this change is to support users migrate from legacy datasets to newer environment and services rule sets and should not be a permanent solution. I think a simple solution of always using a single `.` that's well documented should be good enough.